### PR TITLE
Avoid conflicts when function is implemented in same-named submodule

### PR DESCRIPTION
**Pull request by stefanv**
Supersedes #8

### CODE REVIEW

**Review by jarrodmillman — APPROVED** [review]

File: tests/test_lazy_loader.py [hunk 1]
```diff
@@ -94,3 +94,12 @@ def test_lazy_attach():
...
             assert locls[k] == v
+
+
+def test_attach_same_module_and_attr_name():
+    import fake_pkg
+
+    # Grab attribute twice, to ensure that importing it does not
+    # override function by module
+    assert isinstance(fake_pkg.some_func, types.FunctionType)
+    assert isinstance(fake_pkg.some_func, types.FunctionType)
```
Comment (adrinjalali) [inline tests/test_lazy_loader.py:105]: should we test that `from fake_pkg.some_func import some_func` also works? 

In codebases where this happens, some people do `from fake_pkg import some_func` and some do `from fake_pkg.some_func import some_func` and both work.

Comment (stefanv) [reply tests/test_lazy_loader.py:105]: Good point, I added a test for that.

**Review by adrinjalali — APPROVED** [review]

**Comment by TortoiseHam** [comment]
any updates on a timeline for this change?

**Comment by stefanv** [comment]
Merged, let's see how it goes.

**Comment by TortoiseHam** [comment]
awesome :) Do you anticipate a new rc release with this in the immediate future?

**Comment by jarrodmillman** [comment]
@TortoiseHam Please open an issue if you run into any issues:
https://pypi.org/project/lazy_loader/0.1rc3/

**Comment by jarrodmillman** [comment]
@adrinjalali It would be great if you could test out:
 https://pypi.org/project/lazy_loader/0.1rc3/

**Pull request merged**